src/tasks/WasmAppBuilder/CookieHelper.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+internal static class CookieHelper
+{
+    private static string TypeToChar(Type t)
+    {
+        string? c = t.Name switch
+        {
+            nameof(String) => "I",
```

Comment:
Maybe make `I, L, F, D` into an enum, so the various places in the code are in sync, and handle all the cases.